pkg/clientpool/client.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientpool
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru"
+	"k8s.io/client-go/rest"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// K8sClients is an object of Clients
+var K8sClients *Clients
+
+// Clients is the client pool of k8s client
+type Clients struct {
+	sync.RWMutex
+
+	localConfig *rest.Config
+	clients     *lru.Cache
```

Comment:
Hi @WangXiangUSTC , I think we don't need cache the clients.

As implementation in [kubernetes-dashboard](https://github.com/kubernetes/dashboard/blob/master/src/app/backend/client/manager.go#L415), it will create new clients for each requests.

Reply:
maybe several users will use the chaos dashboard at the same time, and will create the client for each request. I think the cache can improve the performance.